### PR TITLE
fix: create nodeinfo when add task to the nodeinfo and the nodeinfo is not existed

### PR DESCRIPTION
**Pull request by huone1**
Signed-off-by: huone1 <huwanxing@huawei.com>
fix issue #1417

### CODE REVIEW

File: pkg/scheduler/cache/event_handlers.go [hunk 1]
```diff
@@ -59,7 +59,11 @@ func (sc *SchedulerCache) getOrCreateJob(pi *schedulingapi.TaskInfo) *scheduling
 func (sc *SchedulerCache) addTask(pi *schedulingapi.TaskInfo) error {
 	if len(pi.NodeName) != 0 {
 		if _, found := sc.Nodes[pi.NodeName]; !found {
-			return fmt.Errorf("node <%s> does not exist", pi.NodeName)
+			if sc.nodeInformer.Informer().HasSynced() {
```
Comment (k82cn) [inline pkg/scheduler/cache/event_handlers.go:62]: AFAIK, we already sync all informers when starting scheduelr.

Comment (huone1) [reply pkg/scheduler/cache/event_handlers.go:62]: The time sequence of pods and node resources cannot be guaranteed.

Comment (huone1) [reply pkg/scheduler/cache/event_handlers.go:62]: the problem is happened in the starting scheduler period to sync informers data

Comment (huone1) [reply pkg/scheduler/cache/event_handlers.go]: I deleted the judgment."if sc.nodeInformer.Informer().HasSynced" , and  created the nodeinfo structure directly.

Comment (k82cn) [reply pkg/scheduler/cache/event_handlers.go]: what's the impact if there's no node object in node info?

Comment (huone1) [reply pkg/scheduler/cache/event_handlers.go]: if there's no node object in node info, the nodeinfo status will be "NotReady" util the imformer receives the node object 
 to update the status to "Ready". So it is no influence to the volcano schduling and just saves the task info in its nodeInfo.

Comment (k82cn) [reply pkg/scheduler/cache/event_handlers.go]: that's great!

File: pkg/scheduler/api/node_info.go [hunk 2]
```diff
@@ -122,6 +122,10 @@ func (ni *NodeInfo) Ready() bool {
 }
 
 func (ni *NodeInfo) setRevocableZone(node *v1.Node) {
+	if node == nil {
+		return
```
Comment (k82cn) [inline pkg/scheduler/api/node_info.go:126]: let's log a message here, e.g. warning. BTW, will scheduler re-update revocable-zone after that?

Comment (huone1) [reply pkg/scheduler/api/node_info.go:128]: I checked it ， and I found it drop the setRevocableZone in function SetNode and I will add it。
And I add some log in function setNodeGPUInfo, setNodeState, setRevocableZone

**Comment by huone1** [comment]
Close it because the  branch has been confused due to  the merge operation error 
And reopen #1662

**Pull request closed without merging**